src/installer.ts

```diff
@@ -27,31 +27,125 @@ if (!tempDirectory) {
   tempDirectory = path.join(baseLocation, 'actions', 'temp');
 }
 
-export class DotnetCoreInstaller {
+/**
+ * Represents the inputted version information
+ */
+export class DotNetVersionInfo {
+  private fullversion: string;
+  private isExactVersionSet: boolean = false;
+
   constructor(version: string) {
-    if (semver.valid(semver.clean(version) || '') == null) {
-      throw 'Implicit version not permitted';
+    // Check for exact match
+    if (semver.valid(semver.clean(version) || '') != null) {

```

Comment:
Will this still pass on preview versions? i.e. 5.0.0-preview.3

Reply:
Wrote a test ("Resolving a exact preview version works") that covers this.
And yes, it works.